src/io/gff.rs

```diff
@@ -0,0 +1,280 @@
+    }
+
+    /// Write a given GFF record.
+    pub fn write(&mut self, record: Record) -> csv::Result<()> {
+        self.inner.encode(record)
+    }
+}
+
+
+/// A GFF record
+#[derive(RustcEncodable)]
+pub struct Record {
+    sequence: String,
+    source: String,
+    type_name: String,
+    start: u64,
+    end: u64,
+    score: String,
+    strand: String,
+    frame: String,
+    attributes: String,
+}
+
+impl Record {
+    /// Create a new GFF record.
+    pub fn new() -> Self {
+        Record {
+            sequence: "".to_owned(),
+            source: "".to_owned(),
+            type_name: "".to_owned(),
+            start: 0,
+            end: 0,
+            score: ".".to_owned(),
+            strand: ".".to_owned(),
+            frame: "".to_owned(),
+            attributes: "".to_owned(),
+        }
+    }
+
+    /// Sequence of the feature.
+    pub fn sequence(&self) -> &str {
+        &self.sequence
+    }
+
+    /// Source of the feature.
+    pub fn source(&self) -> &str {
+        &self.source
+    }
+
+    /// Type name of the feature.
+    pub fn type_name(&self) -> &str {
+        &self.type_name
+    }
+
+    /// Start position of feature (1-based).
+    pub fn start(&self) -> u64 {
+        self.start
+    }
+
+    /// End position of feature (1-based, not included).
+    pub fn end(&self) -> u64 {
+        self.end
+    }
+
+    /// Score of feature
+    pub fn score(&self) -> Option<u64> {
+        match self.score.as_ref() {
+            "." => None,
+            _ => self.score.parse::<u64>().ok(),
+        }
+    }
+
+    /// Strand of the feature.
+    pub fn strand(&self) -> Option<Strand> {
+        match self.strand.as_ref() {
+            "+" => Some(Strand::Forward),
+            "-" => Some(Strand::Reverse),
+            _ => None,
+        }
+    }
+
+    /// Frame of the feature.
+    pub fn frame(&self) -> &str {
+        &self.frame
+    }
+
+    /// Attribute of feature
+    pub fn attributes(&self) -> &str {
+        &self.attributes
+    }
+    
+    /// Set sequence of feature.
+    pub fn set_sequence(&mut self, sequence: &str) {
+        self.sequence = sequence.to_owned();
+    }
+
+    /// Set source of feature.
+    pub fn set_source(&mut self, source: &str) {
+        self.source = source.to_owned();
+    }
+
+    /// Set type_name of feature.
+    pub fn set_type_name(&mut self, type_name: &str) {
+        self.type_name = type_name.to_owned();
+    }
+
+    /// Set start of feature.
+    pub fn set_start(&mut self, start: u64) {
+        self.start = start;
+    }
+
+    /// Set end of feature.
+    pub fn set_end(&mut self, end: u64) {
+        self.end = end;
+    }
+
+    /// Set score of feature.
+    pub fn set_score(&mut self, score: &str) {
+        self.score = score.to_owned();
+    }
+
+    /// Set strand of feature.
+    pub fn set_strand(&mut self, strand: &str) {
+        self.strand = strand.to_owned();
+    }
```

Comment:
A set_attributes method would take an ``IntoIterator<Item=&str>`` and format it according to the spec.